app/src/main/java/com/nmakademija/nmaakademija/fragment/ScheduleFragment.java

```diff
@@ -70,23 +71,34 @@ public void onFailure(Call<List<ScheduleEvent>> call, Throwable t) {
     }
 
     private void setScheduleItems(List<ScheduleEvent> scheduleEvents) {
-        ArrayList<ScheduleItem> scheduleItems = new ArrayList<>();
-        scheduleItems.addAll(scheduleEvents);
 
+        ScheduleAdapter adapter = new ScheduleAdapter(getContext(), scheduleEvents);
+        scheduleRecyclerView.setAdapter(adapter);
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.US);
         ArrayList<Date> scheduleDates = new ArrayList<>();
-        for (ScheduleEvent scheduleEvent : scheduleEvents) {
-            Date eventDate = scheduleEvent.getDate();
+        List<ScheduleSectionsAdapter.Section> sections = new ArrayList<>();
+        for (int i = 0; i < scheduleEvents.size(); i++) {
+            Date eventDate = scheduleEvents.get(i).getDate();
             Date date = new Date(eventDate.getYear(), eventDate.getMonth(), eventDate.getDate());
+
             if (!scheduleDates.contains(date)) {
```

Comment:
`contains()` complexity on `ArrayList` is `O(N)`. Switch to `HashSet` to have `O(1)` complexity `insert` and `contains`